src/platform/nrfconnect/FactoryDataProvider.cpp

```diff
@@ -42,6 +43,14 @@ CHIP_ERROR FactoryDataProvider<FlashFactoryData>::Init()
     uint8_t * factoryData = nullptr;
     size_t factoryDataSize;
 
+    // Add write protection to the Factory Data flash area
+    constexpr static uint32_t partitionManagerAreaIsWriteProtected = 1;
+    int ret = fprotect_area(PM_FACTORY_DATA_ADDRESS, PM_FACTORY_DATA_SIZE);
+    if(0 != ret || (partitionManagerAreaIsWriteProtected != fprotect_is_protected(PM_FACTORY_DATA_ADDRESS))){
```

Comment:
Is `fprotect_is_protected` needed? Can it happen that `fprotect_area` returns 0 and does nothing?
Also, this will probably not work for the external flash, right? Maybe we should move it to one of the traits that are template arguments of this class?

Reply:
Ok, I read the docs one more time. Fprotect_is_protected isn't needed.